### PR TITLE
remove tox from requirements

### LINKED ISSUE #6: separate requirements
Seems useful for test/dev but we'll never use it for prod. Separate requirements and make prod as slim possible.

https://github.com/uisautomation/sms-webapp/blob/d63400151b0d22bf552d7c1e795e1acff4238d38/requirements.txt#L6

https://github.com/uisautomation/sms-webapp/blob/d63400151b0d22bf552d7c1e795e1acff4238d38/requirements.txt#L9

### PR DESCRIPTION
**Pull request by rjw57**
closes #6

### CODE REVIEW

**Comment by abrahammartin** [comment]
so, the way I understand it, tox was used to execute the test suite inside the container, no? I was suggesting to remove tox from requirements.txt and put in in something like requirements_dev.txt but you have removed it completely. I understand that you think this is now not necessary at all (as removed completely from code) as we have the CircleCI executing tests for each commit. Is my understanding correct?

**Comment by rjw57** [comment]
Sort of. We already have multiple ways of running tox: we can run it directly or via CircleCI. The advantage of having it run via docker-compose is that a local developer can run the tests using Postgres via tox. A developer could already do this via ``docker-compose run manage test`` so that use was marginal.

To add the ability back in, the container would have to install test suite requirements which raises the question: if our container has to install package X to function for our needs, why then is X not a general requirement?

**Comment by rjw57** [comment]
> To add the ability back in, the container would have to install test suite requirements which raises the question: if our container has to install package X to function for our needs, why then is X not a general requirement?

This of course begs the question of whether we want the container built by the Dockerfile we ship to be suitable for production use.

**Comment by rjw57** [comment]
> To add the ability back in, the container would have to install test suite requirements

Actually, that's not true. If we wired up ``setup.py test`` to run tests then we could add ``tox`` to ``tests_require`` in ``setup()``.

**Comment by abrahammartin** [comment]
> Actually, that's not true. If we wired up setup.py test to run tests then we could add tox to tests_require in setup().
I personally think is better to keep the production environment as clean as possible (of requirements/dependencies/external packages/etc that are not directly required to run the prod environment). So for example, django_debug_toolbar is also very useful and I sometimes include it in the dev requirements, but obviously I don't want to include that in the general requirements.txt (for production as well). What do you think?

**Comment by rjw57** [comment]
That makes some sense but then we'd have to think carefully about what goes in this container. Is this container the production one, or one used only for local development?

Having peered at the GCE docs, I'm more convinced than not that we should have all the deployment stuff in a separate repo and have a Dockerfile *there* which checks out this repo and builds a specialised production container. (Like the way the official Python image downloads the Python source rather than using COPY.)

This has the following advantages as I see it:

1. The container can be tagged with a git commit hash and we know absolutely that that container corresponds to that hash. (I believe this is important when we come to deploy since kubernetes rolling deployment upgrades only happen if the image name/tag changes?)
2. Avoiding COPY/ADD means we do not risk accidentally copying a file into the container we do not intend.
3. We can separate deployment configuration from this repo which keeps us honest w.r.t. making this repo "deployment neutral".

So, in summary, I'm edging towards making the Dockerfile/docker-compose.yml we ship in this repo be developer-only. Thoughts?

**Comment by rjw57** [comment]
> I'm edging towards making the Dockerfile/docker-compose.yml we ship in this repo be developer-only.

By which I mean that it also ``pip install``-s a custom requirements.txt file which contains things like tox, django_debug_toolbar, etc, etc.

**Comment by abrahammartin** [comment]
@rjw57 and I agreed that the repo should contain dev/test requirements/Dockerfile/etc and set up a new repo that forks this one to contain production requirements/Dockerfile/docker-compose/etc

**Comment by rjw57** [comment]
OK, will close this PR and merge it into #11 ?
